amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresherTest.java

```diff
@@ -265,7 +266,40 @@ public void testLeaseTableExistsTimesOut() throws Exception {
     }
 
     @Test
-    public void testCreateLeaseTableTimesOut() throws Exception {
+    public void testCreateLeaseTableProvisionedBillingModeIfNotExists() throws Exception {
+        leaseRefresher = new DynamoDBLeaseRefresher(TABLE_NAME, dynamoDbClient, leaseSerializer, CONSISTENT_READS,
+                tableCreatorCallback, LeaseManagementConfig.DEFAULT_REQUEST_TIMEOUT, BillingMode.PROVISIONED);
+
+        when(dynamoDbClient.describeTable(any(DescribeTableRequest.class))).thenReturn(mockDescribeTableFuture);
+        when(mockDescribeTableFuture.get(anyLong(), any()))
+                .thenThrow(ResourceNotFoundException.builder().message("Table doesn't exist").build());
+
+        when(dynamoDbClient.createTable(any(CreateTableRequest.class))).thenReturn(mockCreateTableFuture);
+        when(mockCreateTableFuture.get(anyLong(), any())).thenReturn(null);
+
+        final boolean result = leaseRefresher.createLeaseTableIfNotExists(10L, 10L);
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    public void testCreateLeaseTableIfNotExists() throws Exception {
+        when(dynamoDbClient.describeTable(any(DescribeTableRequest.class))).thenReturn(mockDescribeTableFuture);
+        when(mockDescribeTableFuture.get(anyLong(), any()))
+                .thenThrow(ResourceNotFoundException.builder().message("Table doesn't exist").build());
+
+        when(dynamoDbClient.createTable(any(CreateTableRequest.class))).thenReturn(mockCreateTableFuture);
+        when(mockCreateTableFuture.get(anyLong(), any())).thenReturn(null);
+
+        final boolean result = leaseRefresher.createLeaseTableIfNotExists();
+
+        Assert.assertTrue(result);
+    }
+
```

Comment:
We've only tested the cases where the table doesn't exist (i.e `ddbClient.describeTable()` throws a `ResourceNotFoundException`. Let's add a few other edge cases which can happen:

**DescribeTable**
1. If we throw a `DependencyException`, we should either continue trying to create the table (or throw the exception if we decide to propagate it up as per my earlier comment. I don't have a strong opinion on this since it's existing behavior).
**CreateTable**
2. If we throw a `ResourceNotFoundException`, we should expect to return true.
3. If we throw a `LimitExceededException`, we should propagate up a `ProvisionedThroughputException`.
4. If we throw a `DynamoDBException` we should throw a `DependencyException`.
5. If we throw a `TimeoutException` we should throw a `DependencyException`.